core/src/main/java/com/novoda/noplayer/mediaplayer/forwarder/CompletionInfoForwarder.java

```diff
@@ -0,0 +1,24 @@
+package com.novoda.noplayer.mediaplayer.forwarder;
+
+import android.media.MediaPlayer;
+
+import com.novoda.noplayer.listeners.InfoListeners;
+
+import java.util.HashMap;
+
+class CompletionInfoForwarder implements MediaPlayer.OnCompletionListener {
+
+    private final InfoListeners infoListeners;
+
+    public CompletionInfoForwarder(InfoListeners infoListeners) {
+        this.infoListeners = infoListeners;
+    }
+
+    @Override
+    public void onCompletion(MediaPlayer mp) {
+        HashMap<String, String> keyValuePairs = new HashMap<>();
+        keyValuePairs.put("mp", String.valueOf(mp));
+
+        infoListeners.onNewInfo("onCompletion", keyValuePairs);
```

Comment:
What's the implementation of `onNewInfo`? Is it also holding internally a `Map<String, String>`? 

I wonder if it makes sense to have different `onNewInfo` methods so we can take rid of the keyvalues? Something like `onNewCompletionInfo(mp)` or any other you need in other info forwarders